src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/EntityAIWorkShepherd.java

```diff
@@ -0,0 +1,157 @@
+package com.minecolonies.coremod.entity.ai.citizen.herders;
+
+import com.minecolonies.api.util.constant.ToolType;
+import com.minecolonies.coremod.colony.jobs.JobShepherd;
+import com.minecolonies.coremod.entity.ai.util.AIState;
+import com.minecolonies.coremod.entity.ai.util.AITarget;
+import net.minecraft.entity.passive.EntitySheep;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextComponentTranslation;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.coremod.entity.ai.util.AIState.*;
+
+/**
+ * Created by Asher on 15/9/17.
+ */
+public class EntityAIWorkShepherd extends AbstractEntityAIHerder<JobShepherd, EntitySheep>
+{
+    /**
+     * Experience given per sheep sheared.
+     */
+    protected static final double EXP_PER_SHEEP = 5.0;
+
+    /**
+     * Max amount of animals per Hut Level.
+     */
+    private static final int MAX_ANIMALS_PER_LEVEL = 2;
+
+    /**
+     * Creates the abstract part of the AI.
+     * Always use this constructor!
+     *
+     * @param job the job to fulfill
+     */
+    public EntityAIWorkShepherd(@NotNull final JobShepherd job)
+    {
+        super(job, MAX_ANIMALS_PER_LEVEL);
+        super.registerTargets(
+          new AITarget(SHEPHERD_SHEAR, this::shearSheep)
+        );
+    }
+
+    /**
+     * Decides what job the shephered should switch to, breeding, butchering, or shearing.
+     *
+     * @return the next AIState the shepherd should switch to, after executing this method.
+     */
+    @Override
+    public AIState decideWhatToDo()
+    {
+        setDelay(40);
+
+        final List<EntitySheep> animals = new ArrayList<>(getAnimals());
+
+        if (animals.isEmpty())
+        {
+            setDelay(100);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)